### DISCUSSION TITLE
read.maf: duplicate variants removed?

### DISCUSSION THREAD

**Discussion by complexgenome** [PoisonAlien/maftools#1021] [Q&A] [answered]
Hello there,
I'm using maftools to work on a cohorts' maf. 

I read them and merge using maftools' utility. When the merge is complete I get message that XX number of duplicate variants are removed.

I'd like to know what's causing it to remove. As I understand there are different samples across in the cohort; they may have same SNP but why are they excluded and are considered duplicates?

```

mafs = data.table::rbindlist(l = mafs_datatable, use.names = TRUE, fill = TRUE, idcol = "Source_MAF") 

 mafs_read = maftools::read.maf(maf = mafs) 
```

> Validating 
> --Removed 124759 duplicated variants 
> --Non MAF specific values in Variant_Classification column:   
> START_CODON_SNP   
> DE_NOVO_START_OUT_FRAME   
> DE_NOVO_START_IN_FRAME -
> Silent variants: 29303  -
> Summarizing --
> Possible FLAGS among top ten genes:   
> TTN   MUC16   OBSCN   USH2A 
> -Processing clinical data 
> --Missing clinical data 
> -Finished in 2.638s elapsed (22.1s cpu)
> 

maftools  v2.18.0 
R 4.3.2 (2023-10-31)

Let me know if any other details are needed.

**Comment by PoisonAlien** [comment, marked as answer]
Hi,

Variants are considered duplicates if they occur more than once within the same sample. Here is how they are [identified](https://github.com/PoisonAlien/maftools/blob/4507438c31f7bee5ddde694c6269c4f12c1b13c0/R/validateMaf.R#L26) an removed.

It might be the case that the same variant is being assigned to multiple transcripts - and hence are identified as duplciates. You can turn off with `removeDuplicatedVariants = FALSE` in `read.maf()`

I hope this helps.

**Reply by fanjieqi** [reply]
Thanks a lot~
